EDMarketConnector.py

```diff
@@ -1082,6 +1084,56 @@ def capi_request_data(self, event=None) -> None:  # noqa: CCR001
             play_sound=play_sound
         )
 
+    def capi_request_fleetcarrier_data(self, event=None) -> None:
+        """
+        Perform CAPI fleetcarrier data retrieval and associated actions.
+
+        This is triggered by certain FleetCarrier journal events
+
+        :param event: Tk generated event details.
+        """
+        logger.trace_if('capi.worker', 'Begin')
+        should_return, new_data = killswitch.check_killswitch('capi.fleetcarrier', {})
+        if should_return:
+            logger.warning('capi.fleetcarrier has been disabled via killswitch. Returning.')
+            self.status['text'] = 'CAPI fleetcarrier disabled by killswitch'
+            hotkeymgr.play_bad()
+            return
+
+        if not monitor.cmdr:
+            logger.trace_if('capi.worker', 'Aborting Query: Cmdr unknown')
+            # LANG: CAPI queries aborted because Cmdr name is unknown
+            self.status['text'] = _('CAPI query aborted: Cmdr name unknown')
+            return
+
+        if not monitor.mode:
```

Comment:
I originally excluded the CQC and Multicrew checks as I agree they are irrelevant, but thought a check for a valid commander might be a good idea. Will remove that too though.

Reply:
It's the `monitor.mode` check I was referring to.

You need the `monitor.cmdr` check, else it's going to fail anyway when it can't determine which Cmdr's credentials to use for the query.

Reply:
Ah ok, that makes more sense.